webpack/assets/javascripts/react_app/components/PF4/InlineEdit/InlineEdit.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Spinner,
+  TextInput,
+  TextArea,
+  Text,
+  TextVariants,
+  Button,
+  Split,
+  SplitItem,
+} from '@patternfly/react-core';
+import { TimesIcon, CheckIcon, PencilAltIcon } from '@patternfly/react-icons';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../common/I18n';
+import './inlineEdit.scss';
+
+const InlineEdit = ({ onEdit, value, textArea, attribute }) => {
+  // Tracks input box state
+  const [inputValue, setInputValue] = useState(value);
+  const [editing, setEditing] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+
+  // Setting didCancel to avoid actions from happening after component has been unmounted
+  // see https://overreacted.io/a-complete-guide-to-useeffect/#speaking-of-race-conditions
+  useEffect(() => {
+    let didCancel = false;
+
+    const onSubmit = async () => {
+      if (submitting) {
+        await onEdit(inputValue, attribute);
+        if (!didCancel) {
+          setSubmitting(false);
+          setEditing(false);
+        }
+      }
+    };
+    onSubmit();
+
+    return () => {
+      didCancel = true;
+    };
+  }, [submitting]);
+
+  // Listen for enter and trigger submit workflow on enter
+  useEffect(() => {
+    const listener = event => {
+      if (event.code === 'Enter' || event.code === 'NumpadEnter') {
+        if (editing) setSubmitting(true);
+      }
+    };
+    document.addEventListener('keydown', listener);
+    return () => {
+      document.removeEventListener('keydown', listener);
+    };
+  }, [editing]);
+
+  const onClear = () => {
+    setInputValue(value);
+    setEditing(false);
+  };
+
+  const inputProps = {
+    value: inputValue || '',
+    onChange: v => setInputValue(v),
+  };
+
+  if (submitting) return <Spinner size="sm" />;
+  if (editing) {
+    return (
+      <Split>
+        <SplitItem>
+          {textArea ? (
+            <TextArea {...inputProps} aria-label={`${attribute} text area`} />
+          ) : (
+            <TextInput
+              {...inputProps}
+              type="text"
+              aria-label={`${attribute} text input`}
+            />
+          )}
```

Comment:
I think that this is will introduce quite a bit of complexity to the `InputFactory` component, also I wonder if we are going to have the same approach for pf4? That approach seems pretty coupled with pf3's `FormControl`

The mechanisms here are pretty specific to a text input, as you have to move into an editing state to edit the text and back out. I really envisioned this being used on new pages that want this functionality, rather than an upgrade to pf3 pages.